### PR TITLE
WIP - add java.time.Duration support for jdocs java files #24646

### PR DESCRIPTION
**Pull request by ralphlaude**
I had java.time.Duration support succesfully for most of the classes in the pull request.
I had java.time.Duration support in this class for the method forMax (`akka.actor.FSM.State#forMax`) and i don't know if it is fine.
The test `akka.stream.scaladsl.UnfoldResourceSourceSpec.scala` the method `"use dedicated blocking-io-dispatcher by default"` is broken and i would like to know how to fix it.

### CODE REVIEW

**Comment by raboof** [comment]
@ralphlaude hmm, Jenkins seems fine with the 'use dedicated blocking-io-dispatcher by default' test - does it fail for you on master as well? If so perhaps add a separate ticket about that?

File: akka-docs/src/test/java/jdocs/actor/DependencyInjectionDocTest.java [hunk 1]
```diff
@@ -50,8 +48,9 @@ public static void beforeClass() {
   }
 
   @AfterClass
-  public static void afterClass() throws Exception {
-    Await.ready(system.terminate(), Duration.create("5 seconds"));
+  public static void afterClass() {
+    //Await.ready(system.terminate(), Duration.create("5 seconds"));
```
Comment (raboof) [inline akka-docs/src/test/java/jdocs/actor/DependencyInjectionDocTest.java]: Seems good, remove the commented-out line?

File: akka-docs/src/test/java/jdocs/actor/InitializationDocTest.java [hunk 2]
```diff
@@ -29,8 +28,9 @@ public static void beforeClass() {
   }
 
   @AfterClass
-  public static void afterClass() throws Exception {
-    Await.ready(system.terminate(), Duration.create("5 seconds"));
+  public static void afterClass() {
+    //Await.ready(system.terminate(), Duration.create("5 seconds"));
```
Comment (raboof) [inline akka-docs/src/test/java/jdocs/actor/InitializationDocTest.java]: (idem)

File: akka-docs/src/test/java/jdocs/stream/StreamTestKitDocTest.java [hunk 3]
```diff
@@ -102,7 +99,7 @@ public void pipeToTestProbe() throws Exception {
       .grouped(2)
       .runWith(Sink.head(), mat);
     akka.pattern.PatternsCS.pipe(future, system.dispatcher()).to(probe.ref());
-    probe.expectMsg(FiniteDuration.create(3, TimeUnit.SECONDS),
+    probe.expectMsg(scala.concurrent.duration.FiniteDuration.create(3, TimeUnit.SECONDS),
```
Comment (raboof) [inline akka-docs/src/test/java/jdocs/stream/StreamTestKitDocTest.java]: I think we have an `expectMsg` that takes a Java duration nowadays, right?

**Review by raboof — COMMENTED** [review]
There might be some overlap with #25060 here, but that's probably fine.

Looking pretty good, couple of comments!

File: akka-docs/src/test/java/jdocs/stream/StreamTestKitDocTest.java (hunk 3 above)
Comment (ralphlaude) [reply akka-docs/src/test/java/jdocs/stream/StreamTestKitDocTest.java]: Done

File: akka-docs/src/test/java/jdocs/actor/InitializationDocTest.java (hunk 2 above)
Comment (ralphlaude) [reply akka-docs/src/test/java/jdocs/actor/InitializationDocTest.java]: Done

File: akka-docs/src/test/java/jdocs/actor/DependencyInjectionDocTest.java (hunk 1 above)
Comment (ralphlaude) [reply akka-docs/src/test/java/jdocs/actor/DependencyInjectionDocTest.java]: Done

**Comment by ralphlaude** [comment]
I add one new commit in this pull request. Please check

**Comment by johanandren** [comment]
I just merged the other PR ( #25060 )  to master so now you will have to rebase (and see what calls was already covered by that)

**Comment by ralphlaude** [comment]
I rebase the master, Please check

**Comment by johanandren** [comment]
Looks like something went wrong with that rebase, the diff looks like it is all changes to master since this branch was created or something.

**Comment by raboof** [comment]
@ralphlaude a relatively easy way might be to create a new branch off master, and then `git cherry-pick` only those commits that you want to be part of the PR?

**Comment by ralphlaude** [comment]
i will try to find a way.

**Comment by ralphlaude** [comment]
Hi guys, the new version of this pull request is here (https://github.com/akka/akka/pull/25206).
Please take a look.
Thanks.

**Comment by johanandren** [comment]
Closing this then, as there is a new incarnation

**Pull request closed without merging**